homeassistant/components/tessie/strings.json

```
@@ -528,6 +528,37 @@
     },
     "command_failed": {
       "message": "Command failed, {message}"
+    },
+    "invalid_device": {
+      "message": "The device {device_id} coud not be found or is not part of the Tessie integration"
+    }
+  },
+  "services": {
+    "share": {
+      "name": "Share",
+      "description": "Shares an address, latitude/longitude or video URL to the vehicle.",
```

How should and address be formatted? Or lat/lon? I would argue that these are separate actions which require separate validation

---

Tesla (not Tessie) filter and validate what they allow to be sent to the car server-side. It's a single web service. We can't/don't want to try to validate this on client-side.

---

I would argue that we want to validate something, because who knows in what format it has to be. I'd rather have full services with their own selectors, like a location selector, to make it foolproof

---

Unfortunately, Tesla does not document what it will accept. It would hurt more than it'd help to try to validate this client side. Also, in the future, Tesla could allow us sharing Spotify, Amazon Music, etc URLs. There's no point in trying to validate this on our side. Either Telsa will happily forward it to the car, or it will reject it with an error.

---

yes but if we don't know how it validates, how should the user know? I don't think we should add a bare value, but help guide the user with inputting something that works

---

What about:
```suggestion
      "description": "Shares an address, latitude/longitude (e.g. 48.8583701, 2.2919064) or video URL (e.g. https://www.youtube.com/watch?v=dQw4w9WgXcQ) to the vehicle.",
```

For the address part, my guess is that they run a query to Google Maps or similar, and forward it to the car if it matches something. I just tried `Tour Eiffel`, and it worked, so it's not expecting any format in particular.